com.reprezen.swagedit/src/com/reprezen/swagedit/validation/QuickFixer.java

```diff
@@ -69,16 +71,17 @@ public String getLabel() {
         }
 
         @Override
-        public void processFix(IDocument document, IMarker marker) throws CoreException {
+        public IRegion processFix(IDocument document, IMarker marker) throws CoreException {
             int line = (int) marker.getAttribute(IMarker.LINE_NUMBER);
             try {
                 String indent = getIndent(document, line);
                 // getLineOffset() is zero-based, and imarkerLine is one-based.
                 int endOfCurrLine = document.getLineInformation(line - 1).getOffset()
                         + document.getLineInformation(line - 1).getLength();
                 // should be fine for first and last lines in the doc as well
-                document.replace(endOfCurrLine, 0,
-                        TextUtilities.getDefaultLineDelimiter(document) + indent + "type: object");
+                String replacementText = indent + "type: object";
+                document.replace(endOfCurrLine, 0, TextUtilities.getDefaultLineDelimiter(document) + replacementText);
+                return new Region(endOfCurrLine + 1, replacementText.length());
```

Comment:
@tfesenko Does your "1" hear assume that the line delimiter will be LF? I'll test a CRLF scenario in QA.